### ISSUE TITLE
Self-hosted supergraph not available from server service

### ISSUE THREAD

**Issue by wethekings303** [kamilkisiela/graphql-hive#3772]
While self-hosting graphqlhive the supergraph gives a 404 when trying to reach the supergraph cdn endpoint from multiple sources. 

The server service is available at it's own domain https://hive-server.example.domain and the graphql endpoint is available at https://hive-server.example.domain/graphql with graphiql showing up. The webapp has the option to connect to cdn at https://hive-server.example.domain/artifacts/v1/${TARGET_ID}/supergraph. 

This is for an apollo federation project which I have published 2 schemas to graphql hive with apollo federation spec. I have also created a cdn access token that I provide when trying to get the supergraph from multiple sources such as curl, apollo router with kamilkisiela's image using hive cdn endpoint and hive cdn key variables, and insomnia. For the apollo router I took off the /supergraph at the end per the documentation at https://the-guild.dev/graphql/hive/docs/integrations/apollo-router. When using curl I have a header with X-Hive-CDN-Key using the hive cdn access token. The error on the apollo router is it finds unexpected character "<" with the response of ```<?xml version="1.0" encoding="UTF-8"?>```. 

The main issue is trying to get apollo router to consume the supergraph from graphqlhive. I have the apollo router running and have provided the schema manually in a file but when consuming from graphqlhive I am running into the issues above.

The supergraph is available on minio service and I can reach the artifact in the bucket. I downloaded the artifact and all looks good but when trying to access it through the CDN endpoint provided by the server service It is not accessible. I do see logs of 404 on the server service when trying to access the cdn endpoint for the supergraph.

**Comment by n1ru4l** [comment]
Can you share with us the exact command + configuration that you use for running apollo router with the hive plugin?

**Comment by wethekings303** [comment]
Helm chart overrides from the chart at https://github.com/apollographql/router/tree/dev/helm/chart/router

extraEnvVars:
  - name: APOLLO_ROUTER_CONFIG_PATH
    value: /config/router.yaml
  - name: HIVE_CDN_ENDPOINT
    value: https://hive-server.example.domain/artifacts/v1/${TARGET_ID}
  - name: HIVE_CDN_KEY
    value: '${CDN_KEY_VALUE}'
    
image:
  repository:  ghcr.io/kamilkisiela/graphql-hive/apollo-router
  pullPolicy: IfNotPresent
  # Overrides the image tag whose default is the chart appVersion.
  tag: "75b6c16ef5be6610c912b1cc88ae0587a6af764a"
  
  The router runs with a file mounted via a config map on kubernetes and I can run queries and mutations to the graphql services. When I change from APOLLO_ROUTER_SUPERGRAPH_PATH env var to the HIVE_CDN_ENDPOINT and the HIVE_CDN_KEY vars it fails to start the router.

**Comment by wethekings303** [comment]
https://github.com/kamilkisiela/graphql-hive/tree/main/packages/services/server

The docs for the server service state the s3 public url and s3 endpoint can be different but I had listed the s3 endpoint as the local kubernetes svc such as http://s3.hive.svc.cluster.local:9000. I changed them to be the same and had it externally accessible so that the router could access the artifacts. I also had to remove the s3 reverse proxy after reading the issue at https://github.com/kamilkisiela/graphql-hive/issues/2736. Now the apollo router is running and consuming the supergraph from graphqlhive. Thanks to the team for also adding native composition for apollo federation!
